public/js/logic.js

```diff
@@ -6,35 +6,61 @@ $(document).ready(function(){
 
   // Add a new chest
   $("#add").click(function(){
-    container.append(chest);
-    chestCount++;
+    addChests(1);
   });
 
   // Remove last chest
   $("#remove").click(function(){
-    if (chestCount > 0) {
-      container.children().last().remove();
-      chestCount--;
-    }
+    deleteChests(1);
   });
 
   // Clear all chests
   $("#clear").click(function(){
-    container.empty();
-    chestCount = 0;
+    clearAllChests();
   });
 
-  // Update count when adding, removing, clearing
-  $("#controls button").click(function() {
-    $("#count").val(chestCount);
+  // Remove the chest you click on
+  $("#container").on("click", "div.chest-wrapper", function(){
+    // "on" delegated element events need a static callback function
+    $(this).hide("fast", done());
   });
 
-  // Remove the chest you click on
-  $("#container").on("click", "div", function(){
-    $(this).hide("fast", function(){
-      $(this).remove();
+  function addChests(count) {
+    for(let i = 1; i<=count; i++){
+      container.append(chest);
+      chestCount++;
+    }
+    updateCount();
+  };
+
+  function deleteChests(count) {
+    let loopEnd = Math.min(chestCount, count);
+    for(let i = 1; i<=loopEnd; i++){
+      // Remove current oldest chest from list.
+      container.children().first().remove();
       chestCount--;
-      $("#count").val(chestCount);
-    });
-  });
+    }
+    updateCount();
+  };
+
+  function clearAllChests() {
+    container.empty();
+    chestCount = 0;
+    updateCount();
+  }
+
+  function done(event) {
+    // Event is actually propagated on the .chest element
+    $(event.target).parents('div.chest-wrapper').remove();
+    chestCount--;
+    localStorage.setItem('count', chestCount);
+    $("#count").val(chestCount);
```

Comment:
These should be refactored into update Count right?

Reply:
ah, missed that. Indeed